### ISSUE TITLE
Heterogenous specificity of selector lists in ancestor rules must give an error

### ISSUE THREAD

**Issue by romainmenke** [mrhenry/stylelint-mrhenry#66]
```css
h1,
#foo {
  @media screen {
    color: red;
  }
}

h1 {
  color: green;
}
```

**Comment by romainmenke** [comment]
https://codepen.io/romainmenke/pen/oNVWbao
vs.
https://codepen.io/romainmenke/pen/rNRmxQM
